java/ql/src/semmle/code/java/frameworks/javase/URI.qll

```diff
@@ -0,0 +1,43 @@
+import java
+import semmle.code.java.dataflow.FlowSources
+
+/* Any expresion or call which returns a new URI.*/
+abstract class URICreation extends Top {
+  /*
+   * Returns the host of the newly created URI.
+   *  In the case where the host is specified separately, this returns only the host.
+   *  In the case where the uri is parsed from an input string,
+   *  such as in `URI(`http://foo.com/mypath')`,
+   *  this returns the entire argument passed i.e. `http://foo.com/mypath'.
+   */
+
+  abstract Expr hostArg();
+}
+
+/* An URI constructor expression */
+class URIConstructor extends ClassInstanceExpr, URICreation {
```

Comment:
Use `Uri` not `URI` everywhere (similarly other all-caps words in camel-case names)